R/node-match.R

```diff
@@ -0,0 +1,454 @@
+#' Match patterns to a call
+#'
+#' @description
```

Comment:
This is probably too long for a description and should go in details or separate sections.